### DISCUSSION TITLE
Generic Built-in Auth

### DISCUSSION THREAD

**Discussion by neocybereth** [vercel/next.js#11391] [Help]
Hey guys, love you work and thank you for the amazing framework!

I've noticed when bootstrapping apps with Next I'm finding that rolling my own Auth is taking about the same amount of time as the building of the actual app! Do you guys have any plans to built a pre-rolled generic Auth system for NextJS at at any point?

**Comment by timneutkens** [comment]
> Do you guys have any plans to built a pre-rolled generic Auth system for NextJS at at any point?

There's multiple authentication examples in the examples directory.
Authentication is not really generic between applications and there's many different solutions for it, eg hosted solutions like Auth0 (https://github.com/zeit/next.js/tree/canary/examples/auth0) or authentication against a database, external login like GitHub / Twitter / Facebook and so on.
